app/components/Intercept_Components/InterceptTextBox.tsx

```diff
@@ -8,65 +8,116 @@ export const InterceptTextBox = props => {
   const responseTextValue = props.responseText[props.rowProps.checkbox.requestId] || defaultResponseText;
   const statusCodeValue = props.statusCodes[props.rowProps.checkbox.requestId] || defaultStatusCode;
   const contentTypeValue = props.contentType[props.rowProps.checkbox.requestId] || defaultContentType;
+  let textAreaValue = JSON.stringify(props.responseData[props.rowProps.checkbox.requestId]) ||responseTextValue
 
-  return (
-    <div className="grid-container form">
-      <div className="full-url">
-        <label htmlFor="">URL</label>
-        <a href={props.rowProps.checkbox.url} className="urlText">
-          {props.rowProps.checkbox.url}
-        </a>
+    return (
+      <div className="grid-container form">
+        <div className="full-url">
+          <label htmlFor="">URL</label>
+          <a href={props.rowProps.checkbox.url} className="urlText">
+            {props.rowProps.checkbox.url}
+          </a>
+        </div>
+        <div className="response">
+        <span className="fetch-responsetext" onClick={() => {
+          props.fetchResponse(props.rowProps.checkbox.url,props.rowProps.checkbox.method, props.rowProps.checkbox.requestId )
+          .then( () => {
+            textAreaValue = JSON.stringify(props.responseData[props.rowProps.checkbox.requestId]) ||responseTextValue
+          })
+          .catch( () => {
+            textAreaValue = JSON.stringify(props.responseData[props.rowProps.checkbox.requestId]) ||responseTextValue
+          })
+        }
+        }>X</span>
+          <label htmlFor="">Response Text</label>
+                <textarea
+                name="responseText"
+                className="responseText"
+                defaultValue={textAreaValue }
+                key={textAreaValue}
```

Comment:
Added key attr to force re-render of `textarea` if changing defaultValue after clicking the `fetchResponse (X)` button